tests/test_interpreter.py

```diff
@@ -167,3 +183,34 @@ def test_find_python_interpreter_missing_on_path_raises(monkeypatch):
         find_python_interpreter(interpreter)
         assert "Python Launcher" in str(e)
         assert "on your PATH" in str(e)
+
+
+@pytest.mark.skipif(sys.platform.startswith("win"), reason="Looks for python3")
+def test_fetch_missing_python(monkeypatch):
+    major, minor = _derive_target_python_version()
+
+    python_path = find_python_interpreter(f"{major}.{minor}", fetch_missing_python=True)
+    assert python_path is not None
+    assert f"{major}.{minor}" in python_path or f"{major}{minor}" in python_path
+    assert python_path.endswith("python3")
+
+
+def _derive_target_python_version():
+    minimum_standalone_python_version = list(list_pythons().keys())[-1]
+    minimum_minor_version = int(minimum_standalone_python_version.split(".")[1])
+    major = sys.version_info.major
+    minor = sys.version_info.minor
+
+    # For this test, we want to ask pipx for a version that doesn't exist locally.
+    # Since we test on all supported versions, we can't just pick a specific version to test with.
+    # the requested minor version will always be one less than the minor version currently being tested,
+    # unless the current minor version is the minimum supported version, in which case the requested
+    # minor version will be one more than the current minor version.
+    # ie if the test is running on python 3.9, the requested version will be 3.8
+    # if the test is running on python 3.8, the requested version will be 3.9
+    if minor <= minimum_minor_version:
+        minor = minimum_minor_version + 1
+    else:
+        minor -= 1
+
+    return major, minor
```

Comment:
Does this work in local tests were any number of python versions might already be available?

Reply:
hmmmm... perhaps if you mock-out `shutil.which` to return false either way?